File: python/aula.py
from compromisso import Compromisso

class Aula(Compromisso):
    def __init__(self, id, data, hora, duracao, prioridade, disciplina, nivel):
        super().__init__(id, 'A', data, hora, duracao, prioridade, disciplina)
        
        self.nivel = nivel
        self.calcularPrioridadeFinal()
    
    def calcularPrioridadeFinal(self):
        self.prioridadeFinal = self.prioridade * 2
    
    def getDescricao(self):
        return f"Aula de {self.descricao}"
    
    def isAdiavel(self):
        return False
    
    def getNivel(self):
        return self.nivel

File: C++/src/aula.cpp
#include "../inc/aula.h"

Aula::Aula(){}

Aula::Aula(int id, const Data& data, const Hora& hora, int duracao, int prioridade, const string& descricao, const string& nivel)
    : Compromisso(id, 'A', data, hora, duracao, prioridade, prioridade, descricao), nivel(nivel){}

string Aula::getNivel() const{
    return nivel;
}

string Aula::getDescricao() const{
    return "Aula de " + descricao; 
}

void Aula::calcularPrioridadeFinal(){
    prioridade_final = prioridade * 2;
}
